src/bessel.jl

```diff
@@ -758,3 +758,32 @@ External links: [DLMF](https://dlmf.nist.gov/10.2.6), [Wikipedia](https://en.wik
 See also: [`hankelh2(nu,x)`](@ref SpecialFunctions.hankelh2)
 """
 hankelh2x(nu, z) = besselhx(nu, 2, z)
+
+"""
+    jinc(x)
+
+Bessel function of the first kind divided by `x`.
+Same convention as sinc: ``\\operatorname{jinc}{x} = \\frac{2 \\cdot J_1{\\pi x}}{\\pi x}``.
+Sometimes known as sombrero or besinc function.
+
+External links: [Wikipedia](https://en.wikipedia.org/wiki/Sombrero_function)
+"""
+jinc(x::Number) = iszero(x) ? oftype(x, 1) : _jinc(float(x))
+_jinc(x::Number) = _jinc_core(x)
+
+# below these thresholds we evaluate a fourth order Taylor polynomial
+_jinc_threshold(::Type{Float64}) = 0.001
+_jinc_threshold(::Type{Float32}) = 0.01f0
+
+ # for small arguments, a Taylor series is faster
+@inline _jinc(x::Union{T,Complex{T}}) where {T<:Union{Float32,Float64}} = begin
+    # replace in future with fastabs
+    # https://github.com/JuliaLang/julia/blob/93fb785831dcfcc442f82fab8746f0244c5274ae/base/special/trig.jl#L1057
+    if abs(real(x)) + abs(imag(x)) < _jinc_threshold(T)
+        return evalpoly(x^2, (T(1), -T(π)^2/8, T(π)^4/192))
+    else
+        return _jinc_core(x)
+    end
+end
```

Comment:
This is a fun way to define a function.  Note that the `begin`/`end` block is not needed at all
```julia
@inline _jinc(x::Union{T,Complex{T}}) where {T<:Union{Float32,Float64}} = 
    if abs(real(x)) + abs(imag(x)) < _jinc_threshold(T)
        return evalpoly(x^2, (T(1), -T(π)^2/8, T(π)^4/192))
    else
        return _jinc_core(x)
    end
```
would work just as fine.  You can also use the more standard
```julia
@inline function _jinc(x::Union{T,Complex{T}}) where {T<:Union{Float32,Float64}}
    if abs(real(x)) + abs(imag(x)) < _jinc_threshold(T)
        return evalpoly(x^2, (T(1), -T(π)^2/8, T(π)^4/192))
    else
        return _jinc_core(x)
    end
end
```
For the `evalpoly` issue, you need to use `@evalpoly` instead of `evalpoly`.

Also, for `fastabs` I'd do something like
```julia
function fastabs(x)
    @static if isdefined(Base.Math, :fastabs)
        return Base.Math.fastabs(x)
    else
        # copy definition here....
    end
end
```